tox.ini

```diff
@@ -21,26 +25,51 @@ use_develop = fast: True
 skip_sdist = fast: True
 # Needed on Windows to test data directory creation
 passenv = LOCALAPPDATA
+setenv = COVERAGE_FILE = {env:COVERAGE_FILE:.coverage}
 extras = dev
 # 2023-04-22 see pkgenv ↑
 download = {[pkgenv]download}
 commands =
-    !fast : python -m coverage run -m pytest {posargs:-vv}
-    fast : python -m pytest {posargs:-vv -n auto}
+    !fast : python -m coverage run -m pytest {posargs:-vv --color=yes}
+    fast : python -m pytest {posargs:-vv --color=yes -n auto}
 
-[testenv:coverage{,-html}{,-fail}]
+[testenv:coverage{,38,39,310,311,312}{,-platform}{,-project}{,-html}{,-keep}{,-nofail}]
 depends = py{,38,39,310,311,312}
-# coverage should run on oldest supported Python
-base_python = py38,py39,py310,py311
+# by default, coverage should run on oldest supported Python for testing platform coverage.
+# however, coverage for a particular Python version should match the version used for pytest.
+base_python =
+    coverage: py38,py39,py310,py311,py312
+    coverage38: py38
+    coverage39: py39
+    coverage310: py310
+    coverage311: py311
+    coverage312: py312
 parallel_show_output = True
 extras = dev
 # 2023-04-22 see pkgenv ↑
 download = {[pkgenv]download}
+passenv =
+    COVERAGE_FILE
+    COVERAGE_PLATFORM
+setenv =
+    keep: COMBINE_FLAGS = --keep
+    !nofail: REPORT_FLAGS = --fail-under=100
+    # disable coverage exclusions for Python version to test entire platform
+    {platform,project}: COVERAGE_EXCLUDE_PYTHON_VERSION=disable
+    # disable coverage exclusions for host platform to test entire project
+    project: COVERAGE_EXCLUDE_PLATFORM=disable
+commands_pre =
+    python --version
+    python -c 'if 1: \
+        import os; \
+        print("COVERAGE_FILE", os.environ.get("COVERAGE_FILE", "<not set>")); \
+        print("COVERAGE_PLATFORM", os.environ.get("COVERAGE_PLATFORM", "<not set>")); \
+        print("COVERAGE_EXCLUDE_PYTHON_VERSION", os.environ.get("COVERAGE_EXCLUDE_PYTHON_VERSION", "<not set>")); \
+        print("COVERAGE_EXCLUDE_PLATFORM", os.environ.get("COVERAGE_EXCLUDE_PLATFORM", "<not set>"))'
```

Comment:
I get that these may be useful for debugging purposes, but do they serve a purpose in the long term?

Reply:
Definitely of debatable usefulness now. Because there really isn't any way to introspect what `condition-coverage-plugin` is doing, these were extremely helpful in getting everything working. They may prove useful in the future if things go wrong but we can just add them back if the information would be useful then.